hkmc2/shared/src/test/mlscript/nofib/mate.mls

```diff
@@ -0,0 +1,652 @@
+  append(
+    (if withPiece
+      then append(
+        showPiece([c, k]),
+        if ((k === King) || (k is Pawn and not(capt || prom))) then Nil else "/" :: showSquare(c, sq)
+      )
+      else Nil
+    ),
+    append(
+      maybe("-" :: Nil, cp => "x" :: append(showPiece(cp), "/" :: Nil), mcp),
+      append(
+        showSquare(c, sq_),
+        maybe(Nil, pp => "(" :: append(showPiece(pp), ")" :: Nil), mpp)
+      )
+    ) 
+  )
+
+fun showMoveInFull(a) = showMove(true, a)
+
+fun showMovesAfter(p_, mifs) = if mifs is
+  Nil then Nil
+  MoveInFull(p, sq, d_) :: mifs and p_ is MoveInFull(p_, sq_, _) then
+    append(
+      nofibStringToList(", "),
+      append(
+        showMove(not(eqTuup2(p, p_)) || not(eqTuup2(sq, sq_)), MoveInFull(p, sq, d_)),
+        showMovesAfter(MoveInFull(p, sq, d_), mifs)
+      )
+    )
+
+fun showMoves(mifs) = if mifs is
+  Nil then throw Error("showMoves")
+  mif :: mifs then append(
+    showMoveInFull(mif),
+    showMovesAfter(mif, mifs)
+  )
+
+fun sift(c, bd, ms, sqs) = if sqs is
+  Nil then ms
+  sq :: sqs and
+    onboard(sq) and pieceAt(bd, sq) is
+      None then sift(c, bd, Move(sq, None, None) :: ms, sqs)
+      Some(p_) and
+        colourOf(p_) === c then sift(c, bd, ms, sqs)
+        else sift(c, bd, Move(sq, Some(p_), None) :: ms, sqs)
+    else sift(c, bd, ms, sqs)
+
+fun moveLine(bd, c, sq, inc, cont) = 
+  fun ml(sq, ms) = 
+    let sq_ = inc(sq)
+    if onboard(sq_) and pieceAt(bd, sq_) is
+      None then ml(sq_, Move(sq_, None, None) :: ms)
+      Some(p_) and
+        not(colourOf(p_) === c) then cont(Move(sq_, Some(p_), None) :: ms)
+        else cont(ms)
+    else cont(ms)
+  ms => ml(sq, ms)
+
+// bishopmoves :: Colour -> Square -> Board -> [Move]
+// bishopmoves c sq bd =
+//   ( moveLine bd c sq (\xy -> case xy of {(x,y) -> (x-1,y+1)}) $ (moveLine bd c sq (\xy -> case xy of {(x,y) -> (x+1,y+1)}) $ (moveLine bd c sq (\xy -> case xy of {(x,y) -> (x-1,y-1)}) $ (moveLine bd c sq (\xy -> case xy of {(x,y) -> (x+1,y-1)}) (\x -> x))))) []
+fun bishopmoves(c, sq, bd) =
+  moveLine(
+    bd,
+    c,
+    sq,
+    xy => if xy is [x, y] then [x - 1, y + 1],
+    moveLine(
+      bd,
+      c,
+      sq,
+      xy => if xy is [x, y] then [x + 1, y + 1],
+      moveLine(
+        bd,
+        c,
+        sq,
+        xy => if xy is [x, y] then [x - 1, y - 1],
+        moveLine(
+          bd,
+          c,
+          sq,
+          xy => if xy is [x, y] then [x + 1, y - 1],
+          x => x
+        )
+      )
+    )
+  )(Nil)
+
+// rookmoves :: Colour -> Square -> Board -> [Move]
+// rookmoves c sq bd =
+//   ( moveLine bd c sq (\xy -> case xy of {(x,y) -> (x-1,y)}) $ (moveLine bd c sq (\xy -> case xy of {(x,y) -> (x+1,y)}) $ (moveLine bd c sq (\xy -> case xy of {(x,y) -> (x,y-1)}) $ (moveLine bd c sq (\xy -> case xy of {(x,y) -> (x,y+1)}) (\x -> x))))) []
+fun rookmoves(c, sq, bd) =
+  moveLine(
+    bd,
+    c,
+    sq,
+    xy => if xy is [x, y] then [x - 1, y],
```

Comment:
```suggestion
    case [x, y] then [x - 1, y],
```

Reply:
It seems that the `case` shorthand requires a new line after it:
```fsharp
:fixme
case [x] then x
//│ ╔══[PARSE ERROR] Expected block after `case` keyword; found square bracket section instead
//│ ║  l.5: 	case [x] then x
//│ ╙──     	     ^^^
//│ /!!!\ Uncaught error: scala.MatchError: InfixApp(App(Error(),TyTup(List(Ident(x)))),keyword 'then',Ident(x)) (of class hkmc2.syntax.Tree$InfixApp)

:sjs
case
  [x] then x
//│ JS (unsanitized):
//│ (caseScrut) => {
//│   let first0, x;
//│   if (globalThis.Array.isArray(caseScrut) && caseScrut.length === 1) {
//│     first0 = caseScrut[0];
//│     x = first0;
//│     return x;
//│   } else {
//│     throw new this.Error("match error");
//│   }
//│ }
//│ = [Function (anonymous)]
```

Reply:
Oh you're right, currently it needs to be `case { [x] then x }`. We should fix that at some point.